qiling/extensions/pipe.py

```diff
@@ -48,11 +48,28 @@ def writable(self) -> bool:
     def readable(self) -> bool:
         return True
 
+    def seek(self, offset: int, origin: int) -> int:
+        # Imitate os.lseek
+        raise OSError("Illega Seek")
```

Comment:
`seek` should not be implemented on a non-seekable streams.
Suggesting to remove this and let the caller check for `seekable`. If not seekable, the syscall should return `-ESPIPE` (as opposed to other errors).

Reply:
> `seek` should not be implemented on a non-seekable streams. Suggesting to remove this and let the caller check for `seekable`. If not seekable, the syscall should return `-ESPIPE` (as opposed to other errors).

Then how about other methods like `fileno`, `fstat` and `fnctl`? My idea is that, this imitates how `sys.stdin` really works and a developer doesn't need to care about whether it is `sys.stdin` or some other objects we mocked. And "seek should not be implemented on a non-seekable streams." doesn't make sense to me. Even though `sys.stdin` is not seekable, it still can be seek-ed.

Reply:
Not sure that I follow on this.. The idea is to provide a clean Pythonic interface to the stream.
IMOH, adding all those POSIX / UNIX aliases would mess it up.

Reply:
> Not sure that I follow on this.. The idea is to provide a clean Pythonic interface to the stream. IMOH, adding all those POSIX / UNIX aliases would mess it up.

Yes, `lseek` can be removed, but I insist raising an exception instead of checking `seekable` before calling `seek`.

Reply:
Removed.